scheduler.go

```diff
@@ -1054,3 +1080,70 @@ func WithMonitorStatus(monitor MonitorStatus) SchedulerOption {
 		return nil
 	}
 }
+
+// WithSchedulerMonitor sets a monitor that will be called with scheduler-level events.
+func WithSchedulerMonitor(monitor SchedulerMonitor) SchedulerOption {
+	return func(s *scheduler) error {
+		if monitor == nil {
+			return ErrSchedulerMonitorNil
+		}
+		s.schedulerMonitor = monitor
+		return nil
+	}
+}
+
+// notifySchedulerStarted notifies the monitor that scheduler has started
+func (s *scheduler) notifySchedulerStarted() {
+	if s.schedulerMonitor != nil {
+		s.schedulerMonitor.SchedulerStarted()
+	}
+}
+
+// notifySchedulerShutdown notifies the monitor that scheduler has stopped
+func (s *scheduler) notifySchedulerShutdown() {
+	if s.schedulerMonitor != nil {
+		s.schedulerMonitor.SchedulerShutdown()
+	}
+}
+
+// notifyJobRegistered notifies the monitor that a job has been registered
+func (s *scheduler) notifyJobRegistered(job *Job) {
```

Comment:
Is there a reason you are passing a pointer to the Job interface here? I think this should just be:

```suggestion
func (s *scheduler) notifyJobRegistered(job Job) {
```

Reply:
Thanks @JohnRoesler. You're right  - updated all methods to use `Job` instead of `*Job`. Tests are green. I’ll finish the remaining work and plan to close this PR over the weekend.